form-regex.rst

```diff
@@ -0,0 +1,91 @@
+Using regular expressions
+============================
+
+.. _regex:
+
+Regular Expressions
+---------------------
+
+A regular expression is a special text string for describing a search pattern. Regular expressions find an important use in specifying constraints in your forms. You might want to set constraints on the length, character set allowed and various other fields in the data input by the user. All this can be achieved by using regular expression constraints in your form.
+
+Some basic operations on regular expressions are:
+
+1. **Boolean or**: A vertical bar separates alternatives. For example, ``gray|grey`` can match gray or grey. 
+2. **Grouping Parentheses**: They are used to define the scope and precedence of the operators. For example, ``gray|grey`` and ``gr(a|e)y`` are equivalent patterns which both describe the set of gray or grey.
+3. **Quantification**: A quantifier after a token (such as a character) or group specifies how often that preceding element is allowed to occur. 
+
+    - Square brackets indicate the occurrence of elements within. For example, ``[abc]`` matches either a, b or c character, ``[a-g]`` matches any character from a to g, ``[^abc]`` matches any character except a,b or c.
+    - The question mark indicates zero or one occurrence of the preceding element. For example, ``colou?r`` matches both color and colour.
+    - The asterisk indicates zero or more occurrences of the preceding element. For example, ``ab*c`` matches ac, abc, abbc, abbbc, and so on.
+    - The plus sign indicates one or more occurrences of the preceding element. For example, ``ab+c`` matches abc, abbc, abbbc, and so on, but not ac.
+    - ``{n}``: The preceding item is matched exactly n times. 
+    - ``{min,}``:	The preceding item is matched min or more times. 
+    - ``{min,max}``: The preceding item is matched at least min times, but not more than max times. 
+
+
+For a clear understanding of regular expressions, try these regex online checker tools: https://regex101.com/, https://www.regextester.com/, https://regexr.com/, http://www.regexe.com/.
+
+.. _tips-on-regex:
+
+Tips on using regular expressions
+----------------------------------
+
+- Regular expressions only apply to **strings**. The function used is of the form: ``regex(string value, string expression)`` where it returns the result of regex test on provided value. 
```

Comment:
This is unclear, as you haven't stated where these regexes appear.  (That is, in the "constraint" column on an XForm.)

Also, it looks like the first arg in your examples is `.`  -- You should probably make that explicit here and mention that it represents the input string. Then, what is the result of the function? Is it boolean? What happens if it evaluates to True? or False? or errors out?